crates/cashu/src/nuts/nut18.rs

```diff
@@ -0,0 +1,67 @@
+//! Melting Tokens Onchain
+// https://github.com/cashubtc/nuts/blob/main/18.md
+
+use serde::{Deserialize, Serialize};
+
+use super::nut05::MeltMethodSettings;
+use super::CurrencyUnit;
+use crate::nuts::Proofs;
+use crate::{Amount, Bolt11Invoice};
+
+/// Melt quote request [NUT-18]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MeltQuoteBtcOnchainRequest {
+    /// Bitcoin onchain address to be paid
+    pub amount: Amount,
+    /// Bolt11 invoice to be paid
+    pub address: Bolt11Invoice,
+    /// Unit wallet would like to pay with
+    pub unit: CurrencyUnit,
+}
+
+/// Melt quote response [NUT-18]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MeltQuoteBtcOnchainResponse {
+    /// Quote Id
+    pub quote: String,
+    /// Description
+    pub description: String,
+    /// The amount that needs to be provided
+    pub amount: u64,
+    /// The fee that is required
+    pub fee: u64,
+    /// Whether the the request haas be paid
+    pub paid: bool,
+    /// Unix timestamp until the quote is valid
+    pub expiry: u64,
+}
+
+/// Melt BTC on chain Request [NUT-18]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MeltBtcOnchianRequest {
+    /// Quote ID
+    pub quote: String,
+    /// Proofs
+    pub inputs: Proofs,
+}
+
+impl MeltBtcOnchianRequest {
+    pub fn proofs_amount(&self) -> Amount {
+        self.inputs.iter().map(|proof| proof.amount).sum()
+    }
```

Comment:
you can use fold to avoid map and sum

```suggestion
    pub fn proofs_amount(&self) -> Amount {
        self.inputs
            .iter()
            .fold(Amount::ZERO, |acc, proof| acc + proof.amount)
    }
```

Reply:
I think these are the same from a performance perspective and I prefer the syntax of the map sum. Happy to change it if I am wrong on that though?

https://stackoverflow.com/questions/40441334/is-iter-map-sum-as-fast-as-iter-fold